unit_tests/test_users.py

```diff
@@ -25,13 +27,15 @@ class UsersTest(JenkinsTest):
     def setUp(self):
         super(UsersTest, self).setUp()
         self.useFixture(AptInstalledJenkins(self.fakes))
-        self.fakes.jenkins.scripts[GET_TOKEN_SCRIPT.format("admin")] = "abc\n"
+        self.fakes.jenkins.scripts[GET_LEGACY_TOKEN_SCRIPT.format("admin")] = "abc\n"
         self.users = Users()
 
-    def test_configure_admin_custom_password(self):
+    @mock.patch('charms.layer.jenkins.packages.Packages.jenkins_version')
```

Comment:
For consistency with the approach taken to test the ```Api()``` class, I'd suggest to use the same depedency injection technique:

```
class Users(object):
    __init__(self, packages=None):
        self._packages = packages or Packages()
```

This will avoid having to add mock configuration directives like this in the future, in every possible test that indirectly uses the ```Users()``` class.